sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -204,13 +204,13 @@ export class Recorder {
   /**
    * Sets the matcher for the current recording to the matcher specified.
    */
-  async setMatcher(matcher: Matcher): Promise<void> {
+  async setMatcher(matcher: Matcher, matcherBody?: CustomMatcherBody): Promise<void> {
```

Comment:
If you go with `CustomMatcherOptions` we can call this options instead of matcherBody... 

Also, are options valid for HeaderlessMatcher or BodilessMatcher? If not, it _might_ be confusing when they're ignored but that's totally a nitpick. I wonder if an overload would make sense here?

Reply:
Hmmm, overload.
Great feedback